src/types/resolveStatements.ts

```diff
@@ -388,15 +395,22 @@ function processStatements(
             const r = processStatements(s.statements, sctx, ctx);
             ctx = r.ctx;
 
+            let catchCtx = sctx;
+
             // Process catchName variable for exit code
-            if (initialCtx.vars.has(s.catchName)) {
-                throwError(`Variable already exists: "${s.catchName}"`, s.ref);
+            if (s.catchName != "_") {
+                if (initialCtx.vars.has(s.catchName)) {
+                    throwError(
+                        `Variable already exists: "${s.catchName}"`,
+                        s.ref,
+                    );
+                }
```

Comment:
Let's refactor this code into a helper function and reuse it in this file in numerous places.